src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -311,5 +345,32 @@ private Uri CreateFullEndpointUri(Uri endPoint)
                 }
             }
         }
+
+        private IDictionary<string, string> ParseChallengeData(HttpResponse response)
```

Comment:
None of the logic in this class is unit tested. Every block of code needs to be unit tested, especially code like this.